### ISSUE TITLE
Suggestion: Use permalink instead of slug in Front Matter

### ISSUE THREAD

**Issue by miklb** [voxpelli/node-format-microformat#2]
Just a suggestion for default Front Matter, permalink is a [predefined variable](https://jekyllrb.com/docs/frontmatter/#predefined-global-variables).

**Comment by voxpelli** [comment]
That's maybe something to consider. It would make it easier to figure out the final URL if the formatter takes care of generating it, but would also make it hard to change permalink formats later on.

I think I'm currently more inclined to instead make the expected permalink format configurable, rather than be the hard-coded `%Y/%d` that it currently is: https://github.com/voxpelli/node-format-microformat/blob/8040c422c355cf576bad6b5b361ad86dc3b2cf1f/index.js#L307 Then as long as the configuration matches between the formatter and the actual Jekyll instance then everything would be fine.

(To go a bit over the top: It would be neat if the top Micropub project that glues all three modules together would actually download the Jekyll config and use the data in there to configure the permalink :smile:)

**Comment by miklb** [comment]
I wasn't even thinking about that scenario. I don't use `%Y %d` in my article permalink, so I was thinking for myself not generally.

**Comment by voxpelli** [comment]
I see, you use just `title` in [your permalinks](https://github.com/miklb/miklb.github.com/blob/513eec79915647f51722c45dfae9fa0d9d06b416/_config.yml#L3), hmm...

Won't you run quite a high risk of duplicates then?

Three possible solutions that I'm thinking of:
1. Make the prefix that's sent into `strftime()` configurable and optional. So defined as either a string or false.
2. Make it possible to override the entire `strftime()` part with a custom callback.
3. Make a more extended slug-generator that can take at least a subset of the exact definition that Jekyll accepts.

Number 1 would be absolutely easiest and would enable you to deactivate the prefix for your blog (when also made configurable through the micropub glue)

**Comment by voxpelli** [comment]
A closer look reveals that there's even more parts that may need to be configurable to adapt for both yours and [my permalink](https://github.com/voxpelli/voxpelli.github.com/blob/349c5f3d499a95c893c7f0034e7fa2d2f20ad247/_config.yml#L1) formats.

Will try to give it a look asap when I get some time.

Luckily enough a bad result currently just gives a bad UX, no data corruption 👍

**Comment by miklb** [comment]
Yes, I did realize the issue was with permalinks. I'm just not a fan of date based URLs, but there is a possibility of duplicates.

I think right now I can check if post.permalink is empty & use slug. 

Configurations would be nice, but no rush on my account.

**Comment by miklb** [comment]
Poking around at this some more, for the short term I can set `permalinks: pretty` in `_config.yml` and it will give me a URL. I set my article permalink in the front matter usually anyway, so i can keep the title as URL for most items. 

Obviously long term will be configuration. I might also manually change slug to permalink in my front matter generation for my own uses.

**Comment by voxpelli** [comment]
Permalink formats are now configurable. While it doesn't solve this issue, it's still one step in the right direction.

Related issue: https://github.com/voxpelli/webpage-micropub-to-github/issues/3
